### ISSUE TITLE
image without attr will give bad element with writer.html

### ISSUE THREAD

**Issue by kysko** [jgm/lunamark#79]
Consider:
```
![la lune](lalune.jpg "Voyage *to the* moon")
```
The html writer will output:
```
<p><img src="lalune.jpg" alt="la lune" title="Voyage *to the* moon"</p>
```
with the img end tag ` />` missing.

cause: with the newer parameter `attr` in `writer.html.image` since [july 2020](https://github.com/jgm/lunamark/commit/8e9cdc110d6f12132f7fda7bd9d3c9be274c96f6#diff-bc2e5a5129bef2f66518aa4380bd506d26bfddc2625317d3bbc1bec7f4379111), if `attr` is nil, two nil values w and h will be inserted in the rope.

solution: make w and h empty strings if `attr` is nil

**Comment by Omikhleia** [comment]
Good point. But it also implies the [`ext_link_attributes` test](https://github.com/jgm/lunamark/blob/master/tests/lunamark/ext_link_attributes.test) should have failed, but did not get reported for some reason.

**Comment by kysko** [comment]
I see `shtest` uses `tidy` to "normalize" output... it could have corrected/hidden the error.

**Comment by Omikhleia** [comment]
> solution: make w and h empty strings if attr is nil

(Or generate a different rope in that case, to avoid having empty strings in the serialization)
